### ISSUE TITLE
Accept parser plugin bypasses error handling

### ISSUE THREAD

**Issue by c24w** [restify/node-restify#1619]
- [x] Used appropriate template for the issue type
- [x] Searched both open and closed issues for duplicates of this issue
- [x] Title adequately and _concisely_ reflects the feature or the bug

# Bug Report

## Restify Version

v6.3.4

## Node.js Version

v8

## Expected behaviour

Not acceptable error caused by accept parser plugin can be detected using a standard error listener.

## Actual behaviour

Not acceptable response occurs, but the error listener does not fire.

## Repro case

```
server.use(restify.plugins.acceptParser('JSON'));

server.on('NotAcceptable',  console.log); // Doesn't fire
```
`curl -H 'Accept:application/xml' server-url`

## Cause

The request is terminated [here](https://github.com/restify/node-restify/blob/8f1816621b58637aae6b1fad7a4f6e5a27cc620d/lib/plugins/accept.js#L55-L56).

## Are you willing and able to fix this?

Yes

**Comment by c24w** [comment]
Another note on this: the plugin also always responds using `res.json`, which means it's also not respecting any formatters you have set up.  For example, I might want to serialise the `NotAcceptableError` to text, rather than JSON.

**Comment by DonutEspresso** [comment]
Thanks for the bug report @c24w!

**Comment by c24w** [comment]
Thanks for fixing :)

Do you close the issues when they're released?

**Comment by DonutEspresso** [comment]
Hi - we'll aim to get another 6.x release out this week.

**Comment by DonutEspresso** [comment]
Hi @c24w, we released 6.4.0/7.0.0, both of which include this fix.
